test/hotspot/jtreg/testlibrary_tests/generators/tests/TestGenerators.java

```diff
@@ -0,0 +1,541 @@
+        Asserts.assertEQ(mockGS.uniformInts(-20, -10).restricted(-18, -12).next(), -17);
+
+        // left interval positive
+        mockSource.checkEmpty().enqueueInteger(10, 13, 11);
+        Asserts.assertEQ(mockGS.uniformInts(10, 20).restricted(5, 12).next(), 11);
+
+        // left interval negative
+        mockSource.checkEmpty().enqueueInteger(-12, -9, -11);
+        Asserts.assertEQ(mockGS.uniformInts(-20, -10).restricted(-12, -5).next(), -11);
+
+        // right interval positive
+        mockSource.checkEmpty().enqueueInteger(17, 21, 19);
+        Asserts.assertEQ(mockGS.uniformInts(10, 20).restricted(17, 22).next(), 19);
+
+        // right interval negative
+        mockSource.checkEmpty().enqueueInteger(-20, -16, -19);
+        Asserts.assertEQ(mockGS.uniformInts(-20, -10).restricted(-22, -17).next(), -19);
+
+        mockSource.checkEmpty().enqueueInteger(144);
+        Asserts.assertEQ(mockGS.uniformInts().next(), 144);
+
+        mockSource.checkEmpty();
+
+        int lo = -345555, hi = 11123;
+        var gb = G.uniformInts(lo, hi);
+        for (int i = 0; i < 10_000; i++) {
+            int x = gb.next();
+            Asserts.assertGreaterThanOrEqual(x, lo);
+            Asserts.assertLessThanOrEqual(x, hi);
+        }
+    }
+
+    static void testUniformLongs() {
+        mockSource.checkEmpty().enqueueLong(0, 11, 1).enqueueLong(0, 11, 4);
+        var g0 = mockGS.uniformLongs(0, 10);
+        Asserts.assertEQ(g0.next(), 1L);
+        Asserts.assertEQ(g0.next(), 4L);
+
+        mockSource.checkEmpty().enqueueLong(0, 1, 0).enqueueLong(0, 1, 0);
+        var g1 = mockGS.uniformLongs(0, 0);
+        Asserts.assertEQ(g1.next(), 0L);
+        Asserts.assertEQ(g1.next(), 0L);
+
+        mockSource.checkEmpty().enqueueLong(-1, Long.MAX_VALUE, 10);
+        Asserts.assertEQ(mockGS.uniformLongs(0, Long.MAX_VALUE).next(), 11L);
+
+        mockSource.checkEmpty().enqueueLong(Long.MIN_VALUE, 13, -33);
+        Asserts.assertEQ(mockGS.uniformLongs(Long.MIN_VALUE, 12).next(), -33L);
+
+        mockSource.checkEmpty().enqueueLong(11);
+        Asserts.assertEQ(mockGS.uniformLongs(Long.MIN_VALUE, Long.MAX_VALUE).next(), 11L);
+
+        mockSource.checkEmpty().enqueueLong(10, 29, 17);
+        Asserts.assertEQ(mockGS.uniformLongs(Long.MIN_VALUE, Long.MAX_VALUE).restricted(10L, 28L).next(), 17L);
+
+        mockSource.checkEmpty().enqueueLong(19, 29, 17);
+        Asserts.assertEQ(mockGS.uniformLongs(Long.MIN_VALUE, Long.MAX_VALUE).restricted(10L, 28L).restricted(19L, 33L).next(), 17L);
+
+        mockSource.checkEmpty().enqueueLong(144);
+        Asserts.assertEQ(mockGS.uniformLongs().next(), 144L);
+
+        mockSource.checkEmpty();
+
+        long lo = -344223244511L, hi = 29;
+        var gb = G.uniformLongs(lo, hi);
+        for (int i = 0; i < 10_000; i++) {
+            long x = gb.next();
+            Asserts.assertGreaterThanOrEqual(x, lo);
+            Asserts.assertLessThanOrEqual(x, hi);
+        }
```

Comment:
It might be a bit of work, but it could be nice if we had these range checks for all of the restrictable generators.

Reply:
Well at least those that are safely restrictable without exception.